mempool.go

```diff
@@ -1089,26 +1094,39 @@ func (mp *txMemPool) processOrphans(hash *btcwire.ShaHash) error {
 			enext = e.Next()
 			tx := e.Value.(*btcutil.Tx)
 
-			// Remove the orphan from the orphan pool.
+			// Remove the orphan from the orphan pool.  Current
+			// behavior requires that all saved orphans with
+			// a newly accepted parent are removed from the orphan
+			// pool and potentially added to the memory pool, but
+			// transactions which cannot be added to memory pool
+			// (including due to still being orphans) are expunged
+			// from the orphan pool.
+			//
+			// TODO(jrick): The above described behavior sounds
+			// like a bug, and I think we should investigate
+			// potentially moving orphans to the memory pool, but
+			// leaving them in the orphan pool if not all parent
+			// transactions are known yet.
 			orphanHash := tx.Sha()
 			mp.removeOrphan(orphanHash)
 
 			// Potentially accept the transaction into the
 			// transaction pool.
-			var isOrphan bool
-			err := mp.maybeAcceptTransaction(tx, &isOrphan, true, true)
+			missingParents, err := mp.maybeAcceptTransaction(tx,
+				true, true)
 			if err != nil {
 				return err
 			}
-
-			if !isOrphan {
-				// Generate the inventory vector and relay it.
-				iv := btcwire.NewInvVect(btcwire.InvTypeTx, tx.Sha())
-				mp.server.RelayInventory(iv)
-			} else {
-				mp.removeOrphan(orphanHash)
+			if len(missingParents) != 0 {
```

Comment:
You're right it's a change in behavior.  The orphan was already removed from the orphan pool above (line 1111 in this diff).  It it not added back by the maybeAcceptTransaction (and I added additional comments to document this behavior), so an extra delete is unnecessary.  Additionally, processOrphans is documented to be called on a newly accepted transaction, but in this case the transaction was purged rather than accepted.

However, because the previous behavior of processOrphans was to also remove any orphan transactions which remained orphans, with this change we lose the behavior of recursively removing dependant orphan transactions on the first removed one.  That should be fixed, but I don't think processOrphans is the correct way to handle this (there is no reason to call maybeAcceptTransaction to determine if they are an orphan, because we know at least one parent is missing), so that functionality should be re-added into another function.